### PR TITLE
Add support for Bazel toolchain

### LINKED ISSUE #118: [Feature]: Add bazel as built-in toolchain
### Contact Details

damien.menanteau@gmail.com

### Feature Request

Add [bazel](https://bazel.build/) as a built-in toolchain.
 

### Feature Description

Bazel potentially can work with any language on the 3 platforms macOS/Linux/Windows. This makes it a natural candidate for being defined as a built-in toolchain.

- We already have working examples with java, C++ and Go languages.
- Need to dig a bit deeper on how to make it work with C# language.

### Alternatives

N/A

### Additional Context

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### To Do

- [x] Add example for bazel and java
- [x] Add example for bazel and C++
- [x] Add example for bazel and go
- [x] Add example for bazel and python
- [x] Add bazel as a built-in toolchain in TCR application
- [x] Add possibility to set test-result-dir configuration value to a symbolic link
- [x] Update list of built-in toolchains in README.md

### Parking Lot

- [ ] Add example for bazel and kotlin (TCR built-in bazel toolchain is compatible with kotlin. need to find a working bazel build setup for kotlin)
- [ ] Add example for bazel and C#  (TCR built-in bazel toolchain is compatible with C#. need to find a working bazel build setup for C#)
- [ ] Update dependabot configuration for bazel-related dependency checks. Not available yet in dependabot, cf.  https://github.com/dependabot/dependabot-core/issues/2196

### PR DESCRIPTION
**Pull request by mengdaming**
# Description:
Add bazel as a TCR built-in toolchain.

Fixes #118 

## Type of change
Please tick the appropriate option using [x]

- [ ] Bug fix
- [x] New feature

# Checklist:
Please tick the appropriate options using [x]

- [x] My PR includes tests that cover my code changes.
- [x] Lint and formatter run with no errors
- [x] All new and old tests are passing

### CODE REVIEW

**Comment by philou** [comment]
Do you think the tests in src/languages/<lang>_test.go are required? They seem to be mostly repeating what is declared in the code. Do they catch errors often?

**Review by philou — APPROVED** [review]

**Review by philou — APPROVED** [review]

File: src/xunit/go_junit_ingesters_wrapper.go [hunk 1]
```diff
@@ -48,25 +49,51 @@ func init() {
 func ingestDir(directory string) ([]junit.Suite, error) {
 	var filenames []string
 
-	err := afero.Walk(appFs, directory, func(path string, info os.FileInfo, err error) error {
+	d, errSymLink := evalSymLink(directory)
+	if errSymLink != nil {
+		return nil, errSymLink
+	}
+
+	errWalk := afero.Walk(appFs, d, func(path string, info os.FileInfo, err error) error {
```
[the remark this answers is not in the document]
Comment (mengdaming) [reply src/xunit/go_junit_ingesters_wrapper.go:57]: My bad, my explanation was misleading. Afero is actually a file-system abstraction module. Using it lets you easily plug a real FS (in production code for instance) or other types of FS (such as in-memory FS when running tests).

File: src/xunit/go_junit_ingesters_wrapper_test.go [hunk 2]
```diff
@@ -73,17 +75,19 @@ func assertXunitSampleData(t *testing.T, suites []junit.Suite, nbSamples int) {
...
 }
 
 func Test_ingest_file_wrapper(t *testing.T) {
 	appFs = afero.NewMemMapFs()
 	_ = appFs.Mkdir("build", os.ModeDir)
 	_ = afero.WriteFile(appFs, "build/sample.xml", xunitSample, 0644)
 	suites, err := ingestFile("build/sample.xml")
-	assert.NoError(t, err)
-	assertXunitSampleData(t, suites, 1)
+	if assert.NoError(t, err) {
```
[the remark this answers is not in the document]
Comment (mengdaming) [reply src/xunit/go_junit_ingesters_wrapper_test.go:88]: It's only to keep test results nicer. If an error is detected on line 88, there's no point checking result data themselves in the following assertion. We would also need to enforce the assertion method called in line 89 to check that entries at index 0 are not nil, etc.

**Comment by mengdaming** [comment]
They're useful mainly when adding a new language. Once the language
configuration is stabilised they indeed become less useful.

They could prove handy if someday we implement #178

**Review by aatwi — APPROVED** [review]

**Pull request merged**